test/topology_experimental_raft/test_crash_coordinator_before_streaming.py

```diff
@@ -0,0 +1,131 @@
+    6. Start the old coordinator node
+   
+    Topology operation is expected to fail and cluster is rolled back
+    to previous state.
+
+    | Operation | Coverage |
+    | Decommission | done |
+    | Removenode | done |
+    | Bootstrap | #14804 |
+    | Replace | #14804 |
+
+    2 operations should be covered after issue will be fixed
+
+    """
+    # Decrease the failure detector threshold so we don't have to wait for too long.
+    config = {
+        'failure_detector_timeout_in_ms': 2000
+    }
+    nodes = [await manager.server_add(config=config) for _ in range(3)]
+    cql = manager.get_cql()
+
+    coordinators_ids = await get_coordinator_host_ids(manager)
+    assert len(coordinators_ids) == 1, "At least 1 coordinator id should be found"
+
+    # kill coordinator during decommission
+    coordinator_host = await get_coordinator_host(coordinators_ids[0], manager)
+    other_nodes = [srv for srv in nodes if srv.server_id != coordinator_host.server_id]
+
+    await manager.api.enable_injection(coordinator_host.ip_addr, "crash_coordinator_before_stream", one_shot=True)
+    await manager.decommission_node(server_id=other_nodes[-1].server_id, expected_error="Decommission failed. See earlier errors")
+
+    coordinators_ids = await get_coordinator_host_ids(manager)
+    assert len(coordinators_ids) == 2, "No new record about the coordinator's election"
+    assert coordinators_ids[0] != coordinator_host[1], f"New coordinator wasn't elected {coordinators_ids}"
+    await manager.server_restart(coordinator_host.server_id)
+    await check_token_ring_and_group0_consistency(manager)
+
+    # kill coordinator during removenode
+    await manager.server_add()
+    nodes = await manager.running_servers()
+    coordinators_ids = await get_coordinator_host_ids(manager)
+    coordinator_host = await get_coordinator_host(coordinators_ids[0], manager)
+    other_nodes = [srv for srv in nodes if srv.server_id != coordinator_host.server_id]
+    working_srv_id = other_nodes[0].server_id
+    node_to_remove_srv_id = other_nodes[-1].server_id
+    await manager.server_stop_gracefully(node_to_remove_srv_id)
+    await manager.api.enable_injection(coordinator_host.ip_addr, "crash_coordinator_before_stream", one_shot=True)
+    await manager.remove_node(working_srv_id,
+                              node_to_remove_srv_id,
+                              expected_error="Removenode failed. See earlier errors")
+    
+    coordinators_ids = await get_coordinator_host_ids(manager)
+    assert len(coordinators_ids) == 3, "No new record about the coordinator's election"
+    assert coordinators_ids[0] != coordinator_host[1], f"New coordinator wasn't elected {coordinators_ids}"
+    await manager.server_restart(coordinator_host.server_id)
+    await manager.server_start(node_to_remove_srv_id)
+    await check_token_ring_and_group0_consistency(manager)
+
+    
+
+async def get_coordinator_host_ids(manager: ManagerClient) -> list[str]:
```

Comment:
The following 3 functions look like they can be useful outside this test, no?

Reply:
done